barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.cpp

```diff
@@ -5,65 +5,85 @@ template <class ProverInstances>
+        auto eta = transcript->get_challenge(domain_separator + "_eta");
+        instance->compute_sorted_accumulator_polynomials(eta);
 
-    auto [beta, gamma] = transcript->get_challenges(domain_separator + "_beta", domain_separator + "_gamma");
-    instance->compute_grand_product_polynomials(beta, gamma);
+        // Commit to the sorted withness-table accumulator and the finalized (i.e. with memory records) fourth wire
+        // polynomial
+        witness_commitments.sorted_accum = commitment_key->commit(instance->prover_polynomials.sorted_accum);
+        witness_commitments.w_4 = commitment_key->commit(instance->prover_polynomials.w_4);
 
-    witness_commitments.z_perm = commitment_key->commit(instance->prover_polynomials.z_perm);
-    witness_commitments.z_lookup = commitment_key->commit(instance->prover_polynomials.z_lookup);
+        transcript->send_to_verifier(domain_separator + "_" + commitment_labels.sorted_accum,
+                                     witness_commitments.sorted_accum);
+        transcript->send_to_verifier(domain_separator + "_" + commitment_labels.w_4, witness_commitments.w_4);
+    };
 
-    transcript->send_to_verifier(domain_separator + "_" + commitment_labels.z_perm,
-                                 instance->witness_commitments.z_perm);
-    transcript->send_to_verifier(domain_separator + "_" + commitment_labels.z_lookup,
-                                 instance->witness_commitments.z_lookup);
-    for (size_t idx = 0; idx < NUM_SUBRELATIONS - 1; idx++) {
-        instance->alphas[idx] = transcript->get_challenge(domain_separator + "_alpha_" + std::to_string(idx));
-    }
-    auto vk_view = instance->verification_key->get_all();
-    auto labels = instance->commitment_labels.get_precomputed();
-    for (size_t idx = 0; idx < labels.size(); idx++) {
-        transcript->send_to_verifier(domain_separator + "_" + labels[idx], vk_view[idx]);
-    }
+    const auto grand_products_round = [&](std::shared_ptr<Instance>& instance) {
+        auto& witness_commitments = instance->witness_commitments;
+        auto& commitment_labels = instance->commitment_labels;
+        auto [beta, gamma] = transcript->get_challenges(domain_separator + "_beta", domain_separator + "_gamma");
+        instance->compute_grand_product_polynomials(beta, gamma);
+
+        witness_commitments.z_perm = commitment_key->commit(instance->prover_polynomials.z_perm);
+        witness_commitments.z_lookup = commitment_key->commit(instance->prover_polynomials.z_lookup);
+
+        transcript->send_to_verifier(domain_separator + "_" + commitment_labels.z_perm,
+                                     instance->witness_commitments.z_perm);
+        transcript->send_to_verifier(domain_separator + "_" + commitment_labels.z_lookup,
+                                     instance->witness_commitments.z_lookup);
+    };
+
+    const auto vk_folding_round = [&](std::shared_ptr<Instance>& instance) {
```

Comment:
Any reason these are lambdas instead of normal funcs?

Reply:
it’s an intermediate step that Cody took changing between one big function and many split up functions